### ISSUE TITLE
Bug  no-unnecessary-type-assertion false positive on index access under noUncheckedIndexedAccess

### ISSUE THREAD

**Issue by darcyrush** [oxc-project/tsgolint#1156]
## Repro: oxlint type-aware wrongly removes `!` on index access with `noUncheckedIndexedAccess`

`oxlint --type-aware src.ts` flags `access[0]!` as an unnecessary type assertion
and `--fix` removes it. With `noUncheckedIndexedAccess`, `access[0]` is `Access |
undefined`, so the `!` is required — TypeScript accepts it and errors if removed.

## Minimal Reproduction

https://github.com/darcyrush/oxlint-repro/tree/main

## Reproduce

+ oxlint 1.80.0
+ oxlint-tsgolint 7.0.2001
+ typescript 7.0.2

```bash
npm install -g oxlint oxlint-tsgolint typescript

tsc --noEmit --strict --noUncheckedIndexedAccess src.ts

oxlint src.ts --fix

tsc --noEmit --strict --noUncheckedIndexedAccess src.ts
```

## Observed

```
$ oxlint --type-aware src.ts
src.ts:5:13: error typescript(no-unnecessary-type-assertion): This assertion is unnecessary since it does not change the type of the expression.

$ tsc --noEmit --strict --noUncheckedIndexedAccess  
src.ts:6:24: error TS18048: 'a' is possibly 'undefined'.
```

_LLM assisted_

**Comment by camc314** [comment]
Hi! Thanks for reporting this issue.

In your reproduction repositorym there is no `tsconfig.json`. Passing `--string --noUncheckedIndexedAccess` to `tsc` only configures that invocation; Oxlint doesn't inherit those flags.

Configuring your `tsconfig.json` make both tools use the same settings:

```json
{
  "compilerOptions": {
    "strict": true,
    "noUncheckedIndexedAccess": true
  },
  "include": ["src.ts"]
}
```

Then run:

```sh
tsc --noEmit
oxlint --type-aware src.ts
```

With this configuration, `access[0]!` is preserved. This appears to be a configuration mismatch rather than a rule bug.

If you're still having issue, please make another reproduction and i'll have another look - thanks!
